### ISSUE TITLE
FEATURE REQUEST: Save video as timestamp.format and start in indicator mode

### ISSUE THREAD

**Issue by logileifs** [MaartenBaert/ssr#562]
I would really like it if this small program would start on boot and would be running in background and I could start recording when needed from an indicator in the top panel and stop recording again like that. No need for a big clunky window everytime and the vide output could simply be saved as a timestamp + . + format.

**Comment by logileifs** [comment]
okay I found the `--start-hidden` which works for me to start the app. It is however very annoying to always select a new filename everytime before you start recording. Please add the option to save recording with the name of the current date/timestamp

**Comment by MaartenBaert** [comment]
Check 'separate file per segment'.

You are maybe the 10th or 20th person asking this exact question. I can't believe how people keep requesting the same feature over an over again when said feature already exists. There must be some serious UX flaw with that checkbox somehow but I don't get it, where else would people expect to find that checkbox? Or is the name confusing maybe?

**Comment by logileifs** [comment]
thanks for clearing that up.  'separate file per segment' does in no way tell me anything about the output filename. I guess that's the main UX flaw. separate file per segment sounds to me like you're saving some segments of the video in separate file

**Comment by jabbalaci** [comment]
@MaartenBaert : 'separate file per segment' works strangely and doesn't do what we expect. The problem with that is that it creates a new video even when we pause a video. A video file should be saved and closed when we select 'save recording'.

Let's say I make a tutorial video but installing a software takes too much time and I pause the video. When the installation is ready, resume the video. At the end, when I stop and save the video, I expect to have just one video file, not several segments.

**Comment by lambdacalculator** [comment]
There are two separate issues here: (1) start in the background, (2) allow output filenames to be chosen automatically via timestamp. It sounds like you already found the answer to (1), but I would like to support wholeheartedly your request for (2). Part of the advertising for the program is "Sensible default settings: no need to change anything if you don't want to". Having output filenames automatically chosen for you based on timestamp would be a perfect instance of this, since you would no longer have to think about naming your recordings if you didn't want to. There wouldn't even need to be a UI change: just pre-populate the "Save As" field with a filename chosen by timestamp (current time, time recording started, time recording ended, whatever) and leave it selected. The user could simply type over it if s/he wanted a new name.

**Comment by MaartenBaert** [comment]
I don't like putting the timestamp in the file name field, because then there is no way to figure out next time whether it's a timestamp that needs to be changed, or just something the user entered.

But what I can do is create a separate checkbox for 'separate file per segment' and 'add timestamp'. The reason why it's just a single checkbox right now is because you can't have separate segments without timestamps since that would overwrite the previous file. But I can also just append '-2', '-3' etc when duplicates are detected. I can't really think of a situation where it's useful to intentionally overwrite an existing file.

The 'separate file per segment' checkbox would still be disabled by default, and 'add timestamp' would be enabled by default. Does that make sense?

I can set the default filename to '/home/<username>/Videos/simplescreenrecorder.<ext>', does that make sense? It's unlikely to cause conflicts with anything else.

**Comment by lambdacalculator** [comment]
For comparison, `vokoscreen` automatically gives every output file a name based on the time the recording was started (down to the second), for example `vokoscreen-2017-09-29_09-00-49.mkv`. This file goes into `~/Videos` by default, but that can be changed in the preferences. My choice of screenshooting software, `Shutter`, does something similar, with names like `Selection_073.png` in the `~/Pictures` directory. I've used these programs for several years and never had any problems with either of these schemes. Only occasionally have I renamed files afterward when I needed a more descriptive name. This has the advantage that it places a minimum burden on the user.

However, I do support the idea of giving the user a choice of name, since it can be a burden to rename the output file every time, if one really needs custom names for each recording. It's just that, in this case, it should still be easy if the user doesn't want to be bothered with coming up with a custom name.

In that context, it sounds like your solution with two checkboxes would work well: just once when I start using the program, I could put in the filename `ssr` and check "add timestamp" to get something similar to what `vokoscreen` does, or leave it unchecked and get something similar to what `Shutter` does (since duplicates would be detected). In either case, I would always be free to give any recording a custom name without having to save first and rename later. Maximum flexibility and convenience!
